packages/salesforce-adapter/src/change_validators/record_type_deletion.ts

```diff
@@ -0,0 +1,74 @@
+/*
+*                      Copyright 2022 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import _ from 'lodash'
+import {
+  ChangeError, getChangeData,
+  ChangeValidator, isRemovalChange, ChangeDataType, InstanceElement, isInstanceChange,
+} from '@salto-io/adapter-api'
+import { collections, values } from '@salto-io/lowerdash'
+import { RECORD_TYPE_METADATA_TYPE } from '../constants'
+import { isInstanceOfType } from '../filters/utils'
+
+const { isDefined } = values
+const { awu } = collections.asynciterable
+
+const isRecordTypeChange = async (changedElement: ChangeDataType): Promise<boolean> => (
+  isInstanceOfType(RECORD_TYPE_METADATA_TYPE)(changedElement)
+)
+
+const isTypeDeletion = (changedElement: ChangeDataType): boolean => (
+  changedElement.elemID.idType === 'type'
+)
+
+const isPartOfDeletedType = (instance: InstanceElement, deletedTypes: string[]): boolean => {
+  const fullName = _.get(instance, 'value.fullName')
```

Comment:
```suggestion
  const instanceType = await apiName(instance, true)
```
This should give you what you want, since `apiName` of instance is it's `fullName` (for non CustomObject instances) and by passing the optional parameter `relative=true` it should get you the last segment of the `fullName` field 😇